internal/resources/grafana/common_check_exists_test.go

```diff
@@ -42,14 +42,14 @@ var (
 		func(t *models.NotificationTemplate) string { return t.Name },
 		func(client *goapi.GrafanaHTTPAPI, id string) (*models.NotificationTemplate, error) {
 			resp, err := client.Provisioning.GetTemplate(id)
-			return payloadOrError(resp, err)
+			return payloadOrError[models.NotificationTemplate](resp, err)
```

Comment:
I couldn't make it to automatically infer types, but I am not sure why. Let me know if you have any ideas.